Application/EdFi.Ods.Features/Publishing/Controllers/SnapshotsController.cs

```diff
@@ -0,0 +1,81 @@
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Net;
+using System.Threading.Tasks;
+using EdFi.Ods.Common.Configuration;
+using EdFi.Ods.Common.Constants;
+using EdFi.Ods.Common.Models.Queries;
+using EdFi.Ods.Features.Publishing.Repositories;
+using log4net;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace EdFi.Ods.Features.Publishing.Controllers
+{
+    [ApiExplorerSettings(IgnoreApi = true)]
+    [ExcludeFromCodeCoverage]
+    [ApiController]
+    [Authorize]
+    [Route("data/v3/publishing/snapshots")]
+    public class SnapshotsController : ControllerBase
+    {
+        private readonly IGetSnapshots _getSnapshots;
+        private readonly ILog _logger = LogManager.GetLogger(typeof(SnapshotsController));
+        private readonly bool _isEnabled;
+
+        // NOTE: The optional dependency is necessary to be able to return a 404 Not Found
```

Comment:
I was just following the `DeletesController` implementation from Change Queries as a reference here, so I didn't look at that -- really thought that was intended behavior.

(Worth noting that the `DeletesController` uses `NotFound()` for the result, which will now also return an unintended body in the response.)

Reply:
Yep, perfect example of why we should be cleaning up previous implementations instead of only fixing them going forward, or people will mimic the existing code they find.

Reply:
See comment below: https://github.com/Ed-Fi-Alliance-OSS/Ed-Fi-ODS/pull/229#discussion_r526290455